aea/abstract_agent.py

```diff
@@ -31,6 +31,10 @@ class AbstractAgent(ABC):
     def name(self) -> str:
         """Get agent's name."""
 
+    @abstractproperty
```

Comment:
We should maintain the order first properties then methods throughout for public ones. Private ones can go wherever suitable.